src/services/lessons/lessons.service.js

```diff
@@ -38,53 +43,77 @@ class Lessons {
 			create: 'Failed to create the lesson.',
 			get: 'Failed to get the lesson',
 			find: 'Failed to find lessons.',
-			update: 'Failed to update the lesson.',
+			patch: 'Failed to patch the lesson.',
 			remove: 'Failed to delete the lesson.',
-		}
+			noAccess: 'You have no access.',
+		};
 	}
 
 	setup(app) {
 		this.app = app;
 	}
 
+	clearPermission(lessons) {
+		if (!Array.isArray(lessons)) {
+			lessons = [lessons];
+		}
+		lessons.forEach(l => delete l.permissions);
+		return lessons;
+	}
+
 	async find(params) {
-		const { route } = params;
+		const { route: { courseId }, user } = params;
 
 		try {
 			const lessons = await LessonModel.find({
-				courseId: route.courseId,
+				courseId,
 				deletedAt: { $exists: false },
+			}).populate({
+				path: 'permissions.group',
+				select: 'users',
 			}).select({
 				_id: 1,
 				title: 1,
 				note: 1,
 				visible: 1,
+				permissions: 1,
 				courseId: 1,
 				position: 1,
-			}).lean().exec();
+			}).lean()
+				.exec();
+
+			// todo pagination
+			const lessonsWithAccess = (lessons || []).filter(
+				l => testAccess(l.permissions, user, 'read')
+				|| testAccess(l.permissions, user, 'write'),
```

Comment:
Reicht es hier nicht auf read zu prüfen oder ist der user nicht in read, wenn er in write ist?

Reply:
done